README.md

```diff
@@ -184,6 +185,24 @@ Also see our guides for [builds](./docs/builds.md) and [deployments](./docs/depl
 ### How the integration works
 When a workflow (e.g. GitHub Action) or development event (e.g. pull request, commit, branch) runs, the app receives a webhook from GitHub. The app then extract the issue key from the respective branch/commit/PR and send this information to Jira.
 
+## How the backfill works
+The app is designed to backfill historical data into Jira. Once you have installed and configured the app successfully, it will automatically trigger the backfilling process to update Jira with historical information such as pull requests, deployments, branches, builds, and commits.
+
+1. The backfilling process attempts to connect all branches that fulfill at least one of the following criteria:
+    - The branch name contains the issue key.
+    - The title of the latest pull request associated with the branch contains the issue key.
+    - The last commit message of the branch contains the issue key.
+2. All commits from the default branch will be backfilled. The commit message must contain the Jira issue key.
+3. Only the latest 50 commits from non-default branches will be backfilled.
+4. Unreachable commits (e.g. from deleted branches) will NOT be backfilled.
+5. All pull requests, regardless of their statuses, will be backfilled. The Jira issue key should be included either in 
+   the title of the pull request, in the description of the pull request or in the name of the source branch 
+   of the pull request.
+6. All the builds and deployments data will be backfilled that contain the issue keys. You can check how to include 
```

Comment:
To be specific, it backfills the deployments if there's an issue key in the commits since the last deployment.  Also checks the branch name where the deployment happened to check for an issue key.

The builds are the same